From 053045bc086e460530b5130ab5547ad3af491a81 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:34 -0400
Subject: [ethernet] qlcnic: Reset firmware API lock at driver load time

Message-id: <1408626422-3941-111-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88648
O-Subject: [PATCH RHEL 7.1 110/138] qlcnic: Reset firmware API lock at driver load time
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sony Chacko <sony.chacko@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 2c97e9e2633f3a4a3a301e5071fb0fe0d0d7543b

Some firmware versions fails to reset the lock during
initialization. Force reset firmware API lock during driver
probe to ensure lock availability.

Signed-off-by: Sony Chacko <sony.chacko@qlogic.com>
Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index d0d5a51..2ae3a02 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2371,6 +2371,14 @@ void qlcnic_set_drv_version(struct qlcnic_adapter *adapter)
   qlcnic_fw_cmd_set_drv_version(adapter, fw_cmd);
 }
 
+/* Reset firmware API lock */
+static void qlcnic_reset_api_lock(struct qlcnic_adapter *adapter)
+{
+ qlcnic_api_lock(adapter);
+ qlcnic_api_unlock(adapter);
+}
+
+
 static int
 qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 {
@@ -2473,6 +2481,7 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
  if (qlcnic_82xx_check(adapter)) {
   qlcnic_check_vf(adapter, ent);
   adapter->portnum = adapter->ahw->pci_func;
+  qlcnic_reset_api_lock(adapter);
   err = qlcnic_start_firmware(adapter);
   if (err) {
    dev_err(&pdev->dev, "Loading fw failed.Please Reboot\n"
-- 
1.7.1